api/daemon_test.go

```diff
@@ -33,6 +34,9 @@ func TestStop(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	// Sleep to give time for server to close, as /daemon/stop will return success
+	// before Server.Close() is called.
+	time.Sleep(200 * time.Millisecond)
```

Comment:
Alternatively we could split up `Close` into `closeModules` and `closeServer` and change `daemonStopHandler` to
```go
srv.closeModules()
writeSuccess(w)
src.listener.Close()
```
and change `Close` to:
```go
srv.closeModules()
src.listener.Close()
```